src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useEffect, useState, useRef, DragEvent } from 'react'
+import { Icon, Text } from '@harness/uicore'
+import { useToaster } from '@common/exports'
+import { useStrings } from 'framework/strings'
+import type { uploadHostItem } from '../StepDetails/PdcDetails'
+
+import css from './UploadJSON.module.scss'
+
+interface UploadJSONInterface {
+  setJsonValue: (value: uploadHostItem[]) => void
+}
+
+const UploadJSON = ({ setJsonValue }: UploadJSONInterface) => {
+  const { getString } = useStrings()
+  const { showError } = useToaster()
+  const inputRef = useRef() as React.MutableRefObject<HTMLInputElement>
+  const [fileContent, setFileContent] = useState([] as uploadHostItem[])
+  const [fileName, setFileName] = useState<string>('')
+  const [dropHighlight, setDropHighlight] = useState(false)
+
+  useEffect(() => {
+    setJsonValue(fileContent)
+  }, [fileContent])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'setJsonValue'. Either include it or remove the dependency array. If 'setJsonValue' changes too often, find the parent component that defines it and wrap that definition in useCallback.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
  }, [fileContent, setJsonValue])
```